# GEN711 Final Project
##Pond Microbiome Study

## Members 
- Kendell Chase & Avery Hathaway

## Background
- title of paper 
- authors 
- Summary of original project 
- what data are you using from paper (16s sanger sequencing, whole genome HiC, transcriptome 150bp illumina, etc) 
- what is the goal of your project (e.g. to identify differences in microbiome composition, to identify genes associated with a specific phenotype, etc)

## Methods
- Where data is from
- what type of data is it (150bp illumina, 16s sanger sequencing, etc)
- where are you performing your analysis (RON, Premise, laptop, etc)
- what program did you use first (fastqc, trimmomatic, etc)
- what that program did (trimmed reads, output statistics, etc)
- a paragraph or two at most for each tool
- what did that produce (fastq, bam, vcf, etc)
- what program did you use next (bowtie2, samtools, etc)
- write out whole pipeline ...
- how did you visualize your results (R, python, qiime2 visualizer, etc)

## Findings
- Label figure (Figure1)
- What kind of plot is it? (Venn Diagram)
- What does the plot show (the number of unique genes in each sample, the number of genes shared between samples)
- How did you create the plot (R, python, qiime2 visualizer, etc)
- What was the input to the plot (gene names + counts from bowtie2, etc)
- figure caption formatting (not like the list above)
- Do this for 2 figures

## Citations
